code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -1,116 +1,111 @@
+
+				reagents.reaction(safe_thing, TOUCH)
+				to_transfer = reagents.remove_any(amount_per_transfer_from_this)
+
+				to_chat(user, "<span class='notice'>You transfer [to_transfer] units of the solution.</span>")
+				return
+
+		visible_message("<span class='danger'>[user] squirts something into [C]'s eyes!</span>")
+		reagents.reaction(C, TOUCH)
+
+		var/list/injected = list()
+		for(var/datum/reagent/R in reagents.reagent_list)
+			injected += R.name
+		var/contained = english_list(injected)
+		add_logs(user, C, "squirted", src, "([contained])")
+
+		to_transfer = reagents.trans_to(C, amount_per_transfer_from_this)
+		to_chat(user, "<span class='notice'>You transfer [to_transfer] units of the solution.</span>")
+
+	if(isobj(target))
+		if(!target.reagents)
 			return
 
-		var/trans = 0
-
-		if(ismob(target))
-			if(istype(target , /mob/living/carbon/human))
-				var/mob/living/carbon/human/victim = target
-
-				var/obj/item/safe_thing = null
-				if( victim.wear_mask )
-					if(victim.wear_mask.flags_cover & MASKCOVERSEYES)
-						safe_thing = victim.wear_mask
-				if( victim.head )
-					if(victim.head.flags_cover & MASKCOVERSEYES)
-						safe_thing = victim.head
-				if(victim.glasses)
-					if( !safe_thing )
-						safe_thing = victim.glasses
-
-				if(safe_thing)
-					if(!safe_thing.reagents)
-						safe_thing.create_reagents(100)
-					trans = reagents.trans_to(safe_thing, amount_per_transfer_from_this)
-
-					visible_message("<span class='danger'>[user] tries to squirt something into [victim]'s eyes, but fails!</span>")
-					spawn(5)
-						reagents.reaction(safe_thing, TOUCH)
-
-					to_chat(user, "<span class='notice'>You transfer [trans] units of the solution.</span>")
-					return
-
-
-			visible_message("<span class='danger'>[user] squirts something into [target]'s eyes!</span>")
-			reagents.reaction(target, TOUCH)
-
-			var/mob/living/M = target
-
-			var/list/injected = list()
-			for(var/datum/reagent/R in reagents.reagent_list)
-				injected += R.name
-			var/contained = english_list(injected)
-			M.create_attack_log("<font color='orange'>Has been squirted with [name] by [key_name(user)]. Reagents: [contained]</font>")
-			user.create_attack_log("<font color='red'>Used the [name] to squirt [key_name(M)]. Reagents: [contained]</font>")
-			if(M.ckey)
-				msg_admin_attack("[key_name_admin(user)] squirted [key_name_admin(M)] with [name]. Reagents: [contained] (INTENT: [uppertext(user.a_intent)])")
-			if(!iscarbon(user))
-				M.LAssailant = null
-			else
-				M.LAssailant = user
-
-		trans = reagents.trans_to(target, amount_per_transfer_from_this)
-		to_chat(user, "<span class='notice'>You transfer [trans] units of the solution.</span>")
+		if(reagents.total_volume)
+			if(!target.is_open_container() && !(istype(target, /obj/item/weapon/reagent_containers/food) && !istype(target, /obj/item/weapon/reagent_containers/food/pill)) && !istype(target, /obj/item/clothing/mask/cigarette))

```

Comment:
I get that, I must have forgotten that foods don't qualify as an open container by default, so you have to include logic for them to override the open container check.

My bad.